src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -518,113 +520,180 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
 
-                        assert(embMaskNode != nullptr);
-                        genConsumeReg(embMaskOp);
-                    }
+        assert(embMaskNode != nullptr);
+        genConsumeReg(embMaskOp);
+    }
 
-                    switch (numArgs)
-                    {
-                        case 1:
-                        {
-                            regNumber targetReg  = node->GetRegNum();
-                            GenTree*  rmOp       = node->Op(1);
-                            auto      emitSwCase = [&](int8_t i) {
-                                insOpts newInstOptions = AddEmbRoundingMode(instOptions, i);
-                                genHWIntrinsic_R_RM(node, ins, simdSize, targetReg, rmOp, newInstOptions);
-                            };
-                            regNumber baseReg = internalRegisters.Extract(node);
-                            regNumber offsReg = internalRegisters.GetSingle(node);
-                            genHWIntrinsicJumpTableFallback(intrinsicId, ins, simdSize, lastOp->GetRegNum(), baseReg,
-                                                            offsReg, emitSwCase);
-                            break;
-                        }
-                        case 2:
-                        {
-                            auto emitSwCase = [&](int8_t i) {
-                                insOpts newInstOptions = AddEmbRoundingMode(instOptions, i);
-                                genHWIntrinsic_R_R_RM(node, ins, simdSize, newInstOptions);
-                            };
-                            regNumber baseReg = internalRegisters.Extract(node);
-                            regNumber offsReg = internalRegisters.GetSingle(node);
-                            genHWIntrinsicJumpTableFallback(intrinsicId, ins, simdSize, lastOp->GetRegNum(), baseReg,
-                                                            offsReg, emitSwCase);
-                            break;
-                        }
+    instruction ins = HWIntrinsicInfo::lookupIns(node);
 
-                        default:
-                        {
-                            unreached();
-                        }
-                    }
-                }
-                else
-                {
-                    // We shouldn't encounter embedded masking for non table-driven intrinsics
-                    assert((embMaskNode == nullptr) && (embMaskOp == nullptr));
+    // The intrinsic table uses the most expressive form of the instruction
+    // to make lowering, lsra, and other phases simpler. In some cases the
+    // most expressive form is EVEX only and so we want to fallback to an
+    // older instruction when we either don't need EVEX or can't use it.
 
-                    // There are a few embedded rounding intrinsics that need to be emitted with special handling.
-                    genNonTableDrivenHWIntrinsicsJumpTableFallback(node, lastOp);
-                }
+    switch (ins)
```

Comment:
This switch block is the actual "peephole" added. It does the last minute check for if we've got an EVEX only instruction but aren't actually using any of the features that would require it to be EVEX only (namely embedded broadcast or masking).

If EVEX isn't strictly required we fallback to the baseline instruction. That instruction will allow typical cases to emit the smaller VEX encoding instead but also allows the EVEX encoding to still be used in the case XMM16-XMM31 is used,